File: desktop/src/main/java/ru/kidesoft/desktop/domain/entity/order/Ticket.java
package ru.kidesoft.desktop.domain.entity.order;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.time.ZonedDateTime;

@Data
@Builder
@AllArgsConstructor
public class Ticket {
    private Integer id;
    private String number;
    private StatusType status;
    private String zone;
    private Integer seatNumber;
    private Integer rowSector;
    private Float amount;
    private String showName;
    private String ageLimit;
    private ZonedDateTime dateTime;
}

File: adapter/src/main/kotlin/ru/kidesoft/ticketplace/adapter/domain/order/Ticket.kt
package ru.kidesoft.ticketplace.adapter.domain.order

import java.time.ZonedDateTime

data class Ticket(
    var id: Int,
    var number: String,
    var status: StatusType,
    var zone: String,
    var seatNumber: Int,
    var rowSector: Int,
    var amount: Float,
    var showName: String,
    var ageLimit: String,
    var dateTime: ZonedDateTime
) : Cloneable {
    override fun clone(): Ticket {
        return Ticket(id, number, status, zone, seatNumber, rowSector, amount, showName, ageLimit, ZonedDateTime.from(dateTime))
    }
}

